src/structures/InteractionCollector.js

```diff
@@ -51,23 +80,26 @@ class MessageComponentInteractionCollector extends Collector {
     this.total = 0;
 
     this.empty = this.empty.bind(this);
-    this._handleChannelDeletion = this._handleChannelDeletion.bind(this);
-    this._handleGuildDeletion = this._handleGuildDeletion.bind(this);
-    this._handleMessageDeletion = this._handleMessageDeletion.bind(this);
+
+    if (this.message) {
+      this._handleMessageDeletion = this._handleMessageDeletion.bind(this);
+      this.client.on(Events.MESSAGE_DELETE, this._handleMessageDeletion);
+    }
+    if (this.channel) {
+      this._handleChannelDeletion = this._handleChannelDeletion.bind(this);
+      this.client.on(Events.CHANNEL_DELETE, this._handleChannelDeletion);
+    }
+    if (this.guild) {
+      this._handleGuildDeletion = this._handleGuildDeletion.bind(this);
+      this.client.on(Events.GUILD_DELETE, this._handleGuildDeletion);
+    }
```

Comment:
Small looks nit, alongside a request to increment the max listeners per client.on call :D

```suggestion
    if (this.message) {
      this._handleMessageDeletion = this._handleMessageDeletion.bind(this);
      this.client.on(Events.MESSAGE_DELETE, this._handleMessageDeletion);
    }

    if (this.channel) {
      this._handleChannelDeletion = this._handleChannelDeletion.bind(this);
      this.client.on(Events.CHANNEL_DELETE, this._handleChannelDeletion);
    }

    if (this.guild) {
      this._handleGuildDeletion = this._handleGuildDeletion.bind(this);
      this.client.on(Events.GUILD_DELETE, this._handleGuildDeletion);
    }
```

Reply:
I don't think it's necessary to do it per call? The maxListeners property determines how many listeners can be bound to each event, not in total. This is only binding one listener to each event, so incrementing once should be sufficient.

I'll move the increment call to the top though.